easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1463,6 +1463,15 @@ def _parse_dependency(self, dep, hidden=False, build_only=False):
             if dependency['full_mod_name'].split('/')[-1].startswith('.'):
                 dependency['hidden'] = True
 
+            name_version = dependency['short_mod_name'].split('/')
+            if 'name' not in dependency['external_module_metadata']:
+                dependency['name'] = name_version[0]
+            if 'version' not in dependency['external_module_metadata']:
+                if len(name_version) > 1:
+                    dependency['version'] = name_version[1]
+                else:
+                    dependency['version'] = ''
```

Comment:
@akesandgren This is still quite hacky, and will have unintended side effects.

We better make `HierarchicalMNS` compatible with Cray toolchains, where there is no info available on the toolchain compiler (since there's no dedicated toolchain component for it), see https://github.com/akesandgren/easybuild-framework/pull/17